go/vt/vttablet/tabletserver/schema/engine.go

```diff
@@ -670,8 +670,13 @@ func (se *Engine) RegisterVersionEvent() error {
 	return se.historian.RegisterVersionEvent()
 }
 
-// GetTableForPos returns a best-effort schema for a specific gtid
-func (se *Engine) GetTableForPos(tableName sqlparser.IdentifierCS, gtid string) (*binlogdatapb.MinimalTable, error) {
+// GetTableForPos returns a best-effort schema for a specific gtid. If it cannot get
+// the table schema for the gtid, it returns the latest table schema available in the
+// database (updating the cache entry). If the table is not found in the cache, it will
+// reload the cache from the database in case the table was created after the last schema
+// reload or the cache has not yet been initialized. This function makes the schema
+// cache a read-through cache for VReplication purposes.
```

Comment:
This is fantastic. Asking to check: is there a legitimate scenario where the code would be asked again and again for a table that does not exist _yet_, and which would cause the cache to reload too much (as in creating excessive load on the server / locking / whatever)?

Reply:
Let's say that you start a MoveTables workflow and mention a table that doesn't exist.... the workflow will go into the error state. And IF the error is considered ephemeral we'll retry again in 5 seconds by default. And repeat.

I'll think this through and do some testing to see if it's a potential issue in practice and if so, how to better deal with it. As a final backstop, we do have `--vreplication_max_time_to_retry_on_error` although that's disabled by default.

Reply:
The `NoSuchTable` error is considered unRecoverable and thus we do not retry:

https://github.com/vitessio/vitess/blob/eb22cfb0db109aa9f5491aa2684414412b0d420c/go/vt/vttablet/tabletmanager/vreplication/controller.go#L262-L277

https://github.com/vitessio/vitess/blob/eb22cfb0db109aa9f5491aa2684414412b0d420c/go/vt/vttablet/tabletmanager/vreplication/utils.go#L195

So the potentially problematic scenario that came to my mind should not be an issue.